src/Nethermind/Nethermind.Blockchain/Synchronization/EthSyncPeerPool.cs

```diff
@@ -389,6 +389,9 @@ private async Task RefreshPeerInfo(PeerInfo peerInfo, CancellationToken token)
                             }
                         }
                     }
+
+                    delaySource.Dispose();
```

Comment:
move to finally - I can see at least one return that would omit it, also any exception and so on